src/kemal/param_parser.cr

```diff
@@ -71,11 +72,17 @@ module Kemal
         next unless upload
 
         filename = upload.filename
+        name = upload.name
 
         if !filename.nil?
-          @files[upload.name] = FileUpload.new(upload)
+          if name.ends_with?("[]")
+            @all_files[name] ||= [] of FileUpload
+            @all_files[name] << FileUpload.new(upload)
+          else
+            @files[name] = FileUpload.new(upload)
+          end
```

Comment:
Oh, interesting. I hadn't even thought about adding to different collections based on parameter name. My vision was that both ways for accessing should work for the same parameters. Either as different collections, or different ways to query the same collection.
I suppose this works as well, as long as `[]` is a clear indicator for multiple files and there could not be multiple files for a parameter name that doesn't end in `[]`. It's my understanding that the `[]` suffix is just a convention, though. So applications could be confronted with multiple files for the same parameter name which doesn't end with `[]`. In that case, the approach taken here would be insufficient.